src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -33,6 +33,31 @@ var invariant = require('invariant');
 
 const isArray = Array.isArray;
 
+function formatUnexpectedArgument(arg) {
+  var type = typeof arg;
+  if (type !== 'object') {
+    return type;
+  }
+  var displayName = arg.constructor && arg.constructor.name || type;
+  var keys = Object.keys(arg);
+  if (keys.length > 0 && keys.length < 20) {
+    return `${displayName} (keys: ${keys.join(', ')})`;
+  }
+  return displayName;
+}
+
+function validateCallback(callback, callerName) {
+  if (typeof callback !== 'function') {
+    invariant(
+      false,
+      '%s(...): Expected the last optional `callback` argument to be a ' +
+      'function. Instead received: %s.',
+      callerName,
+      formatUnexpectedArgument(callback)
+    );
+  }
+}
```

Comment:
Even if they're not direct equivalents, if `validateCallback` is meant to validate a callback specifically for enqueued callbacks it would make sense for it to live alongside the rest of the queue logic. Not a big deal though.